src/modules/accommodations/accommodations.service.ts

```diff
@@ -52,4 +60,78 @@ export class AccommodationsService {
       totalPages: Math.ceil(totalCount / limit),
     };
   }
+
+  async createAccommodation(
+    createDto: CreateAccommodationDto,
+  ): Promise<Accommodation> {
+    const { amenities, categoryId, ...accommodationData } = createDto;
+    console.log(createDto);
+    const allAmenities = await this.amenitiesService.getAllAmenities();
+    const resolvedAmenities = amenities
+      ? allAmenities.filter((amenity) => amenities.includes(amenity.id))
+      : [];
+
+    const allCategories = await this.categoryService.getAllCategories();
+    const resolvedCategory = allCategories.find(
+      (category) => category.id === categoryId,
+    );
+
+    if (!resolvedCategory) {
+      throw new NotFoundException(`Category with ID ${categoryId} not found`);
+    }
+
+    const newAccommodation = this.accommodationRepository.create({
+      ...accommodationData,
+      amenities: resolvedAmenities,
+      category: resolvedCategory,
+    });
+
+    return this.accommodationRepository.save(newAccommodation);
+  }
+
+  async getAccommodationById(id: string): Promise<Accommodation> {
+    const accommodation = await this.accommodationRepository.findOne({
+      where: { id },
+      relations: ['amenities', 'category', 'owner'],
+    });
+
+    if (!accommodation) {
+      throw new NotFoundException(`Accommodation with ID ${id} not found`);
+    }
+
+    return accommodation;
+  }
+
+  async updateAccommodation(
+    id: string,
+    updateDto: UpdateAccommodationDto,
+  ): Promise<Accommodation> {
+    const { amenities, categoryId, ...updateData } = updateDto;
+
+    const accommodation = await this.getAccommodationById(id);
+    if (amenities) {
+      const allAmenities = await this.amenitiesService.getAllAmenities();
+      accommodation.amenities = allAmenities.filter((amenity) =>
+        amenities.includes(amenity.id),
+      );
+    }
+    if (categoryId) {
+      const allCategories = await this.categoryService.getAllCategories();
+      const category = allCategories.find((cat) => cat.id === categoryId);
+
+      if (!category) {
+        throw new NotFoundException(`Category with ID ${categoryId} not found`);
+      }
+
+      accommodation.category = category;
+    }
+
+    Object.assign(accommodation, updateData);
+
+    return this.accommodationRepository.save(accommodation);
+  }
+
+  async deleteAccommodation(id: string): Promise<void> {
+    await this.accommodationRepository.delete(id);
```

Comment:
you should make sure that the record got deleted. When you delete from typeorm it returns object that has "affected" property. You can check if affected is 0 and throw notfoundexception because accommodation does not exist.

here is an example from what I wrote in user service:
```js
  async deleteUser(userId: string): Promise<void> {
    const deleteResult = await this.usersRepository.deleteUser(userId);

    if (deleteResult.affected === 0) {
      throw new NotFoundException(`User with id ${userId} not found`);
    }
  }
```